comms/src/outbound_message_service/outbound_message_service.rs

```diff
@@ -312,11 +292,11 @@ mod test {
             .chain(outbound_message.message_envelope.body_frame())
             .result()
             .to_vec();
-        assert!(signature.verify_challenge(&node_identity.public_key, &challenge));
+        assert!(signature.verify_challenge(&node_identity.identity.public_key, &challenge));
         // Check Encryption
         assert_eq!(message_envelope_header.flags, MessageFlags::ENCRYPTED);
         let ecdh_shared_secret =
-            RistrettoPublicKey::shared_secret(&dest_sk.clone(), &node_identity.public_key).to_vec();
+            RistrettoPublicKey::shared_secret(&node_identity.secret_key, &dest_peer.public_key).to_vec();
```

Comment:
The new line is incorrect. This line tested that the destination node was able to generate the shared secret and decrypt the message.

Reply:
Ah ok, that will need to be worked on - the new PeerNodeIdentity the does not have the secret key (we wont have the destination peers secret key)

Reply:
Ah ok, I've changed it to use the destination node's secret key - the new PeerNodeIdentity doesn't have a secret key (we won't know the dest node's secret key)

Reply:
Perfect